pkg/scheduler/plugins/gpushare/pod_handler.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gpushare
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/api/v1/resource"
+
+	apiext "github.com/koordinator-sh/koordinator/apis/extension"
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+)
+
+func (n *nodeDeviceCache) onPodAdd(obj interface{}) {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		klog.V(4).Infof("pod cache add failed to parse, obj %T", obj)
+		return
+	}
+
+	podRequest, _ := resource.PodRequestsAndLimits(pod)
+	if !isGPUPod(podRequest) {
+		klog.V(5).Infof("pod cache add skip non GPU pod, podName: %v", pod.Name)
+		return
+	}
+
+	devicesAllocation, err := apiext.GetDeviceAllocations(pod)
+	if err != nil {
+		klog.Errorf("failed to get %v device allocation, err: %v", pod.Name, err)
+		return
+	}
+
+	n.assume(devicesAllocation[schedulingv1alpha1.GPU], pod.Spec.NodeName)
+	klog.V(4).InfoS("pod cache added", "pod", klog.KObj(pod))
+}
+
+func (n *nodeDeviceCache) onPodUpdate(oldObj, newObj interface{}) {
+	return
+}
+
+func (n *nodeDeviceCache) onPodDelete(obj interface{}) {
+	var pod *corev1.Pod
+	switch t := obj.(type) {
+	case *corev1.Pod:
+		pod = t
+	case cache.DeletedFinalStateUnknown:
+		var ok bool
+		pod, ok = t.Obj.(*corev1.Pod)
+		if !ok {
+			klog.V(4).Infof("pod cache delete failed to parse, obj %T", obj)
+			return
+		}
+	default:
+		return
+	}
+
+	podRequest, _ := resource.PodRequestsAndLimits(pod)
+	if !isGPUPod(podRequest) {
+		klog.V(5).Infof("pod cache delete skip non GPU pod, podName: %v", pod.Name)
+	}
+
+	devicesAllocation, err := apiext.GetDeviceAllocations(pod)
+	if err != nil {
+		klog.Errorf("failed to get %v device allocation, err: %v", pod.Name, err)
+		return
+	}
+
+	n.unAssume(devicesAllocation[schedulingv1alpha1.GPU], pod.Spec.NodeName)
+	klog.V(4).InfoS("pod cache deleted", "pod", klog.KObj(pod))
+}
+
+// assume is used to update deviceUsed when there is a new pod created
+func (n *nodeDeviceCache) assume(allocations []*apiext.DeviceAllocation, nodeName string) {
+	info := n.getNodeDevice(nodeName)
+	if info == nil {
+		klog.Errorf("node device cache not found, nodeName: %v", nodeName)
+		return
+	}
+
+	info.lock.Lock()
+	defer info.lock.Unlock()
+
+	if len(allocations) != 0 {
+		for _, allocation := range allocations {
+			info.updateDeviceUsed(schedulingv1alpha1.GPU, allocation, true)
```

Comment:
even if you want to update only schedulingv1alpha1.GPU, you should add "if deviceType == schedulingv1alpha1.GPU";  anyway,  you should handle every deviceType